text/0000-scaling-use-cases.md

```diff
@@ -0,0 +1,115 @@
+
+Another challenge is to scan the transactions of the global wallet. The most common stop criteria for the scanner is the GAP Limit, which stops the scanner when `N` addresses in a row are empty. An address is empty when it has no transactions.
+
+To illustrate the problem, let's say the GAP Limit is three. If Charlie and Dave have not made any deposits, addresses #3, #4, and #5 will be empty, and the wallet will not find the deposit made by Eve in address #6.
+
+A temporary solution might be to increase the GAP Limit, but the issue will likely happen again in the future.
+
+Another solution might be to artificially generate transactions every three users.
+
+<!-- ##### -->
+<!-- ##### -->
+
+## Global Wallet with Derivation Paths
+
+This case is similar to the Global Wallet. But, instead of associating one index per user, it associates one derivation path per user. For instance:
+
+```
+derivation_path | user
+----------------|----------
+  m/44'/0'/0'/0 | Alice
+  m/44'/0'/0'/1 | Bob
+  m/44'/0'/0'/2 | Charlie
+  m/44'/0'/0'/3 | Dave
+  m/44'/0'/0'/4 | Eve
+```
+
+Notice that each user has one, and only one, derivation path.
+
+Each derivation path leads to an unlimited number of enumerable addresses. Thus, each user has addresses #0, #1, #2, #3, and so on. Notice that Alice's address #0 is different from Bob's address #0 because they had different derivation paths.
+
+This approach has a similar scaling issue of the Global Wallet. The wallet will have to keep track of all these derivation paths and their addresses.
+
+Another challenge is to scan the derivation paths. The most common stop criteria for the scanner is the GAP Limit M/N, which stops the scanner when `M` derivation paths in a row are empty. It considers that a given derivation path is empty when its first `N` addresses are empty.
+
+## One Wallet Per User
+
+This case differs from the previous ones because there will be multiple wallets running.
+
+First of all, there are multiple ways to associate a wallet with a user. To illustrate, some ways are: (i) one seed per user, (ii) one global seed with one passphrase per user, or (iii) one global seed with one derivation path per user.
+
+The challenges here are the same regardless the way the wallet is associated with a user.
+
+The naive implementation here is to run one wallet per user. But it would consume a lot of resources (CPU and memory) and it is hard to scale to millions of users.
+
+A better implementation is to run a pool of wallets that automatically starts and stops wallets on demand. For instance, when a request is made to a wallet that is not running, the pool starts that wallet automatically. When a wallet is idle for too long, it is automatically stopped.
+
+Such pool would consume resources according to the number of concurrent users (instead of total users). It can easily scale up through the addition of more servers to the pool.
+
+## Full-length Wallet Service
```

Comment:
I think we should add a note explaining that this option is not currently available as it is being tested internally